Hunting Queries/SecurityEvent/powershell_newencodedscipts.yaml

```
@@ -0,0 +1,43 @@
+id: 4e78daf1-8bba-4b5d-8a8b-c75fe9bbc2d9
+name: New PowerShell scripts encoded on the commandline
+description: |
+  'Identify and decode new encoded powershell scripts this week versus previous 14 days'
+requiredDataConnectors:
+  - connectorId: SecurityEvents
+    dataTypes:
+      - SecurityEvent
+tactics:
+  - Execution
+  - CommandAndControl
+query: |
+
+  let starttime = 21d;
+  let midtime = 14d;
+  let endtime = 7d;
+  let ProcessCreationEvents=() {
+  let processEvents=SecurityEvent
+  | where EventID==4688
+  | project  TimeGenerated, ComputerName=Computer,AccountName=SubjectUserName,AccountDomain=SubjectDomainName,
+    FileName=tostring(split(NewProcessName, '\\')[-1]),
+  ProcessCommandLine = CommandLine, 
+  InitiatingProcessFileName=ParentProcessName,InitiatingProcessCommandLine="",InitiatingProcessParentFileName="";
+  processEvents};
+  let encodedPSScripts = 
+  ProcessCreationEvents 
+  | where TimeGenerated >= ago(midtime)
+  | where FileName =~ "powershell.exe"
+  | where ProcessCommandLine contains "-encodedCommand";
+  encodedPSScripts
+  | where TimeGenerated > ago(endtime)
+  | summarize count() by ProcessCommandLine
```

Add a min(TimeGenerated) as a timestamp?
```suggestion
  | summarize count(), timestamp = min(TimeGenerated) by ProcessCommandLine
```

---

Done, thanks for catching.